aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -547,7 +547,7 @@ public AWSAppSyncClient build() {
                     mServerUrl = appSyncJsonObject.getString("ApiUrl");
                     mRegion = Regions.fromName(appSyncJsonObject.getString("Region"));
 
-                    if (mUseClientDatabasePrefix) {
+                    if (mUseClientDatabasePrefix && (mClientDatabasePrefix == null || StringUtils.isBlank(mClientDatabasePrefix))) {
```

Comment:
```suggestion
                    if (mUseClientDatabasePrefix && StringUtils.isBlank(mClientDatabasePrefix)) {
```

You don't need the null check, [it's already in `StringUtils.isBlank(...)`](https://github.com/aws-amplify/aws-sdk-android/blob/main/aws-android-sdk-core/src/main/java/com/amazonaws/util/StringUtils.java#L260-L266).